### PR TITLE
Update keymaster types

### PR DESCRIPTION
**Pull request by wtachau**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/madrobby/keymaster/blob/master/keymaster.js#L86
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/keymaster/keymaster-tests.ts [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
 
 // simple key binding with scope
-key('shift+b', 'aScope', (keyboardEvent, keymasterEvent) => {
-});
-
-// possible combinations of filtering
-key.filter({target: { tagName: 'propertytest'}});
-key.filter({srcElement: { tagName: 'propertytest'}});
-
+key("shift+b", "aScope", (keyboardEvent, keymasterEvent) => {});
```
Comment (BendingBender) [inline types/keymaster/keymaster-tests.ts:5]:
```suggestion
key("shift+b", "aScope", (keyboardEvent, keymasterEvent) => {
    keyboardEvent; // $ExpectType KeyboardEvent
    keymasterEvent; // $ExpectType KeymasterEvent
});
```

File: types/keymaster/keymaster-tests.ts [hunk 2]
```diff
@@ -1,14 +1,12 @@
-
-
 // simple key binding without scope
-key('shift+a', function (keyboardEvent: KeyboardEvent, keymasterEvent: KeymasterEvent) {
-});
+key("shift+a", function (keyboardEvent: KeyboardEvent, keymasterEvent: KeymasterEvent) {});
```
Comment (BendingBender) [inline types/keymaster/keymaster-tests.ts:2]:
```suggestion
key("shift+a", (keyboardEvent: KeyboardEvent, keymasterEvent: KeymasterEvent) => {
    keyboardEvent; // $ExpectType KeyboardEvent
    keymasterEvent; // $ExpectType KeymasterEvent
});
```

File: types/keymaster/keymaster-tests.ts [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-// possible combinations of filtering
-key.filter({target: { tagName: 'propertytest'}});
-key.filter({srcElement: { tagName: 'propertytest'}});
-
+key("shift+b", "aScope", (keyboardEvent, keymasterEvent) => {});
+
+// This re-sets the default filter method: https://www.npmjs.com/package/keymaster#filter-key-presses
+const element = new HTMLElement();
+key.filter = event => {
+    const tagName = (event.target || event.srcElement)!.tagName;
```
Comment (BendingBender) [inline types/keymaster/keymaster-tests.ts:10]:
```suggestion
    event; // $ExpectType FilterEvent
    const tagName = (event.target || event.srcElement)!.tagName;
```

**Review by BendingBender — APPROVED** [review]

File: types/keymaster/keymaster-tests.ts (hunk 2 above)
Comment (BendingBender) [inline types/keymaster/keymaster-tests.ts:2]:
```suggestion
key("shift+a", (keyboardEvent: KeyboardEvent, keymasterEvent: KeymasterEvent) => {
    keyboardEvent; // $ExpectType KeyboardEvent
    keymasterEvent; // $ExpectType KeymasterEvent
});
```

File: types/keymaster/keymaster-tests.ts (hunk 1 above)
Comment (BendingBender) [inline types/keymaster/keymaster-tests.ts:5]:
```suggestion
key("shift+b", "aScope", (keyboardEvent, keymasterEvent) => {
    keyboardEvent; // $ExpectType KeyboardEvent
    keymasterEvent; // $ExpectType KeymasterEvent
});
```

File: types/keymaster/keymaster-tests.ts (hunk 3 above)
Comment (BendingBender) [inline types/keymaster/keymaster-tests.ts:10]:
```suggestion
    event; // $ExpectType FilterEvent
    const tagName = (event.target || event.srcElement)!.tagName;
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

File: types/keymaster/keymaster-tests.ts [hunk 4]
```diff
@@ -1,14 +1,19 @@
-
-
 // simple key binding without scope
-key('shift+a', function (keyboardEvent: KeyboardEvent, keymasterEvent: KeymasterEvent) {
+key("shift+a", (keyboardEvent: KeyboardEvent, keymasterEvent: KeymasterEvent) => {
```
Comment (BendingBender) [inline types/keymaster/keymaster-tests.ts:2]: Sorry, overlooked this:
```suggestion
key("shift+a", (keyboardEvent, keymasterEvent) => {
```

**Review by BendingBender — CHANGES_REQUESTED** [review]

**Review by BendingBender — APPROVED** [review]

**Comment by BendingBender** [comment]
Ready to merge

**Comment by wtachau** [comment]
Ready to merge

**Pull request merged**